### PR TITLE
Infrastructure/3412 ga4 stories

### PR DESCRIPTION
**Pull request by eugene-manuilov**
## Summary

Addresses issue #3412

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks good, just a few tweaks and this one needs QA brief filled in.

File: assets/js/modules/analytics/components/settings/SettingsForm.js [hunk 1]
```diff
@@ -54,8 +54,9 @@ export default function SettingsForm() {
 	const isGA4Connected = useSelect( ( select ) => select( CORE_MODULES ).isModuleConnected( 'analytics-4' ) );
 	const setupFlowMode = useSelect( ( select ) => select( STORE_NAME ).getSetupFlowMode() );
 	const hasExistingTag = useSelect( ( select ) => select( STORE_NAME ).hasExistingTag() );
-	const profileID = useSelect( ( select ) => select( STORE_NAME ).getProfileID() );
+	const { accountID, profileID } = useSelect( ( select ) => select( STORE_NAME ).getSettings() );
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/settings/SettingsForm.js]: Updated.

File: assets/js/modules/analytics/components/settings/SettingsForm.js [hunk 2]
```diff
@@ -88,13 +89,13 @@ export default function SettingsForm() {
 				</div>
 			) }
 
-			{ ( isGA4Enabled && isGA4Connected ) && (
+			{ ( isGA4Enabled && isGA4Connected && Array.isArray( ga4Properties ) && ga4Properties.length > 0 ) && (
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/settings/SettingsForm.js]: Updated.

File: assets/js/modules/analytics/components/settings/SettingsForm.js [hunk 3]
```diff
@@ -88,13 +89,13 @@ export default function SettingsForm() {
...
 				<div className="googlesitekit-setup-module__inputs googlesitekit-setup-module__inputs--collapsed">
 					<GA4PropertySelect label={ __( 'Google Analytics 4 Property', 'google-site-kit' ) } />
 				</div>
 			) }
 
-			{ ( isGA4Enabled && SETUP_FLOW_MODE_UA === setupFlowMode && isGA4Connected ) && (
+			{ ( isGA4Enabled && isGA4Connected && Array.isArray( ga4Properties ) && ga4Properties.length === 0 ) && (
```
[the remark this answers is not in the document]
Comment (eugene-manuilov) [reply assets/js/modules/analytics/components/settings/SettingsForm.js]: Updated.

**Review by tofumatt — APPROVED** [review]

**Pull request merged**